Fix x86_64-gnu-llvm-15 CI tests

Kobzol opened the pull request:
The CI script was broken - if there was a test failure in the first command chain (inside the `if`), CI would not report the failure.

It happened because there were two command chains separated by `&&` in the script, and since `set -e` doesn't exit for chained commands, if the first chain has failed, the script would happily continue forward, ignoring any test failures.

This could be fixed e.g. by adding some `|| exit 1` to the first chain, but I suppose that the `&&` chaining is unnecessary here anyway.

Reported [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/test.20failure.20didn't.20stop.20CI).

Fixes: https://github.com/rust-lang/rust/issues/116867

Kobzol committed:
Fix x64_64-gnu-llvm-15 CI tests

There were two command chains separated by `&&` in the script, and since `set -e` doesn't exit for chained commands, if the first chain has failed, the command would happily continue forward, ignoring any test failures.

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6571240246/job/17850019265) [(plain)](https://github.com/rust-lang/rust/commit/07abf5bd2d2ec015a2b13b1d980efe8fc056890f/checks/17850019265/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
<missing>      2 weeks ago      /bin/sh -c #(nop)  LABEL org.opencontainers.…   0B        
<missing>      2 weeks ago      /bin/sh -c #(nop)  ARG LAUNCHPAD_BUILD_ARCH     0B        
<missing>      2 weeks ago      /bin/sh -c #(nop)  ARG RELEASE                  0B        

<botocore.awsrequest.AWSRequest object at 0x7fa09237f490>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
https://ci-caches.rust-lang.org/docker/645a42c1f063660d7d1b8cceb7e5f0e4270e8fe67ce6c7a408521655a418b77aa3ecbdca6e49eb2f3a943ea0c0cb13c9992c5b6a1062f47d03d9e9b066c228ce
sha256:70637e77b6e56146e57898eae8f29de680b85b88f4f8a8a5f6e72d2dddee26b1
```

</details>

Kobzol committed:
Fix `span_use_eq_ctxt` test

The stage0 compiler does not know about the lint yet, so ignore the test on stage1.

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6571521481/job/17850831556) [(plain)](https://github.com/rust-lang/rust/commit/8cd09be2faa8cda853167f03a71210b0bae5da17/checks/17850831556/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
<missing>      2 weeks ago      /bin/sh -c #(nop)  LABEL org.opencontainers.…   0B        
<missing>      2 weeks ago      /bin/sh -c #(nop)  ARG LAUNCHPAD_BUILD_ARCH     0B        
<missing>      2 weeks ago      /bin/sh -c #(nop)  ARG RELEASE                  0B        

<botocore.awsrequest.AWSRequest object at 0x7f145b01b710>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
https://ci-caches.rust-lang.org/docker/645a42c1f063660d7d1b8cceb7e5f0e4270e8fe67ce6c7a408521655a418b77aa3ecbdca6e49eb2f3a943ea0c0cb13c9992c5b6a1062f47d03d9e9b066c228ce
sha256:9c37d8d85963793bc62768333c04c3c71067162bc7c95de7a80da08bd6b0af74
---

---- [ui] tests/ui-fulldeps/internal-lints/span_use_eq_ctxt.rs stdout ----
diff of stderr:

1 error: use `.eq_ctxt()` instead of `.ctxt() == .ctxt()`
+   --> $DIR/span_use_eq_ctxt.rs:13:5
3    |
3    |
4 LL |     s.ctxt() == t.ctxt()

6    |
7 note: the lint level is defined here
-   --> $DIR/span_use_eq_ctxt.rs:5:9
---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/internal-lints/span_use_eq_ctxt.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/internal-lints/span_use_eq_ctxt" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/internal-lints/span_use_eq_ctxt/auxiliary" "-Z" "unstable-options"
stdout: none
--- stderr -------------------------------
error: use `.eq_ctxt()` instead of `.ctxt() == .ctxt()`
   |
   |
LL |     s.ctxt() == t.ctxt() //~ ERROR use `.eq_ctxt()` instead of `.ctxt() == .ctxt()`
   |
note: the lint level is defined here
  --> /checkout/tests/ui-fulldeps/internal-lints/span_use_eq_ctxt.rs:6:9
   |
```

</details>

lqd commented:
fixes #116867

Kobzol commented:
Ok, this should now be ready.

@rustbot ready

lqd commented:
This looks good to me and matches the fix Eric suggested in that zulip topic. The `#[cfg(bootstrap)]` is also in a rust test, not a script, so will also show up in the next [beta bump procedure](https://forge.rust-lang.org/release/process.html#master-bootstrap-update-t-2-day-tuesday).

Mark-Simulacrum commented:
@bors r+ rollup=iffy

bors commented:
:pushpin: Commit c2524bc6899a1a7feebda032dd26f751504e94e5 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum c2524bc6899a1a7feebda032dd26f751504e94e5 -->
<!-- homu: {"type":"Approved","sha":"c2524bc6899a1a7feebda032dd26f751504e94e5","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit c2524bc6899a1a7feebda032dd26f751504e94e5 with merge 9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3...
<!-- homu: {"type":"BuildStarted","head_sha":"c2524bc6899a1a7feebda032dd26f751504e94e5","merge_sha":"9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/6601047043/job/17932638787)
Approved by: Mark-Simulacrum
Pushing 9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/6601047043/job/17932638787"},"merge_sha":"9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3"} -->

rust-timer commented:
Finished benchmarking commit (9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3): [comparison URL](https://perf.rust-lang.org/compare.html?start=97a28940626caf771d332525315079ded8a36e47&end=9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=97a28940626caf771d332525315079ded8a36e47&end=9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=97a28940626caf771d332525315079ded8a36e47&end=9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | 2.1%  | [2.1%, 2.1%]   | 1     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -4.0% | [-4.5%, -3.0%] | 3     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=97a28940626caf771d332525315079ded8a36e47&end=9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=97a28940626caf771d332525315079ded8a36e47&end=9e3f784eb2c7c847b6c3578b373c0e0bc9233ca3&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 631.011s -> 632.258s (0.20%)
**Artifact size**: 301.90 MiB -> 301.94 MiB (0.01%)
